src/Component/spec/Filter/NumericRangeFilterSpec.php

```diff
@@ -0,0 +1,198 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\Sylius\Component\Grid\Filter;
+
+use PhpSpec\ObjectBehavior;
+use Sylius\Component\Grid\Data\DataSourceInterface;
+use Sylius\Component\Grid\Data\ExpressionBuilderInterface;
+use Sylius\Component\Grid\Filtering\FilterInterface;
+
+final class NumericRangeFilterSpec extends ObjectBehavior
+{
+    function it_implements_a_filter_interface(): void
+    {
+        $this->shouldImplement(FilterInterface::class);
+    }
+
+    function it_does_nothing_when_there_is_no_data(DataSourceInterface $dataSource): void
```

Comment:
This should check that `$dataSource->restrict()` was not called, like in the ExistsFilter for example: https://github.com/Sylius/SyliusGridBundle/blob/5d3f702947f17502e0d6e173276086c1d8ebcf9b/src/Component/spec/Filter/ExistsFilterSpec.php#L30-L35

For some reason, the same was done for the MoneyFilter, but it does not seem right:
https://github.com/Sylius/SyliusGridBundle/blob/5d3f702947f17502e0d6e173276086c1d8ebcf9b/src/Component/spec/Filter/MoneyFilterSpec.php#L34-L42

/cc @Sylius/core-team